tools/querytee/response_comparator.go

```diff
@@ -116,6 +118,60 @@ func (s *SamplesComparator) Compare(expectedResponse, actualResponse []byte) (Co
 	return ComparisonSuccess, nil
 }
 
+var errorEquivalenceClasses = [][]*regexp.Regexp{
+	{
+		// Invalid expression type for range query: MQE and Prometheus' engine return different error messages.
+		// Prometheus' engine:
+		regexp.MustCompile(`invalid parameter "query": invalid expression type "range vector" for range query, must be Scalar or instant Vector`),
+		// MQE:
+		regexp.MustCompile(`invalid parameter "query": query expression produces a range vector, but expression for range queries must produce an instant vector or scalar`),
```

Comment:
Should MQE be changed to be the same for this one? (and later promql can be fixed to be more descriptive)

Reply:
We've intentionally decided to provide clearer error messages where we can in MQE, even if that means diverging from Prometheus' engine.

Reply:
Yep, but I'm wondering if we should be the same to ease testing, and open PRs upstream to fix up wording. Nonetheless, not a blocker here :-)